src/Analyzers/Core/CodeFixes/NamespaceSync/AbstractNamespaceSyncCodeFixProvider.CustomFixAllProvider.cs

```diff
@@ -0,0 +1,90 @@
+                    FixAllScope.Project => await fixAllContext.GetAllDiagnosticsAsync(fixAllContext.Project).ConfigureAwait(false),
+                    FixAllScope.Solution => await GetSolutionDiagnosticsAsync(fixAllContext).ConfigureAwait(false),
+                    _ => default
+                };
+
+                if (diagnostics.IsDefaultOrEmpty)
+                    return null;
+
+                return new MyCodeAction(
+                    FixAllContextHelper.GetDefaultFixAllTitle(fixAllContext),
+                    cancellationToken => FixAllByDocumentAsync(
+                        fixAllContext.Project.Solution,
+                        diagnostics,
+                        cancellationToken));
+
+                static async Task<ImmutableArray<Diagnostic>> GetSolutionDiagnosticsAsync(FixAllContext fixAllContext)
+                {
+                    var diagnostics = ImmutableArray.CreateBuilder<Diagnostic>();
+
+                    foreach (var project in fixAllContext.Solution.Projects)
+                    {
+                        var projectDiagnostics = await fixAllContext.GetAllDiagnosticsAsync(fixAllContext.Project).ConfigureAwait(false);
+                        diagnostics.AddRange(projectDiagnostics);
+                    }
+
+                    return diagnostics.ToImmutable();
+                }
+            }
+
+            private static async Task<Solution> FixAllByDocumentAsync(
+                Solution solution,
+                ImmutableArray<Diagnostic> diagnostics,
+                CancellationToken cancellationToken)
+            {
+                // Use documentId instead of tree here because the
+                // FixAsync call can modify more than one document per call. The
+                // important thing is that the fix works on fixing the namespaces in a single document,
+                // but references in other documents will be updated to be correct. Id will remain
+                // across this mutation, but lookup via SyntaxTree directly will not work because
+                // the tree won't be the same.
+                var documentIdToDiagnosticsMap = diagnostics
+                    .GroupBy(diagnostic => diagnostic.Location.SourceTree)
+                    .Where(group => group.Key is not null)
+                    .ToImmutableDictionary(group => solution.GetRequiredDocument(group.Key!).Id, group => group.ToImmutableArray());
+
+                var newSolution = solution;
+
+                foreach (var (documentId, diagnosticsInTree) in documentIdToDiagnosticsMap)
+                {
+                    var document = newSolution.GetRequiredDocument(documentId);
+                    newSolution = await FixAsync(document, diagnosticsInTree, cancellationToken).ConfigureAwait(false);
+                }
+
+                return newSolution;
```

Comment:
please report progress as you're doing your work.  i can walk you through how to do this.

Reply:
Do you have an example of how to do this?